lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -84,6 +84,36 @@ ::mlir::LogicalResult mlir::tt::ttnn::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttnn::ReshapeOp::verify() {
```

Comment:
In the file `ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/reshape/reshape_op.cpp` I found this validation

```
void Reshape::validate(const std::vector<Tensor> &input_tensors) const {
```
I believe we should have more or less same validations on our side. Now the question is whether should we implement these and maintain in sync with ttnn lib, or should link against their lib and call this validation. Probably a topic for tech sync. For now, I'd implement these as part of TTNN op verify function.

Reply:
That validate was added for `reshape_on_device`. `ttnn::reshape`doesn't reference that function, but regradless if I want to call it I would have to pull in `tt-metal` lib into `tt-mlir`, which I would like to avoid doiing just for this function.
But I agree with your comment regarding validation, we should figure out who, where and how we should validate...